src/main/java/com/elixr/poc/rest/validator/PurchaseValidator.java

```diff
@@ -0,0 +1,25 @@
+package com.elixr.poc.rest.validator;
+import com.elixr.poc.Constant.ErrorConstants;
+import com.elixr.poc.rest.request.PurchaseRequest;
+import java.util.ArrayList;
+import java.util.List;
+
+public class PurchaseValidator {
```

Comment:
I would suggest using the spring boot validation framework to validate incoming request

Reply:
updated